gcc-8 is upset by this "const" in the JOINT_VISITOR_INIT() instanciation.
Droping it should not harm, until it is understood what it wrong.

--- src/multibody/joint/joint-basic-visitors.hxx~	2018-09-18 14:52:22.278553130 +0200
+++ src/multibody/joint/joint-basic-visitors.hxx	2018-09-18 14:52:50.059659284 +0200
@@ -105,7 +105,7 @@
   struct JointCalcAbaVisitor : public fusion::JointVisitor<JointCalcAbaVisitor>
   {
     typedef boost::fusion::vector< Inertia::Matrix6 &,
-                                    const bool > ArgsType;
+                                    bool > ArgsType;
 
     JOINT_VISITOR_INIT(JointCalcAbaVisitor);
 
